### ISSUE TITLE
ntpd fails to start on alpine ARM image 

### ISSUE THREAD

**Issue by matclab** [c3os-io/c3os#20]
The logs say:
```
Jan  6 12:00:20 monitor daemon.warn openrc[1809]: WARNING: clock skew detected!
Jan  6 12:00:21 monitor daemon.err acpid[1830]: /dev/input/event0: No such file or directory
Jan  6 12:00:21 monitor daemon.err /etc/init.d/ntpd[1855]: start-stop-daemon: failed to start `
/usr/sbin/ntpd'
Jan  6 12:00:21 monitor daemon.err /etc/init.d/ntpd[1837]: ERROR: ntpd failed to start
…
Jan  6 12:00:30 monitor auth.info sshd[1887]: User c3os not allowed because account is locked
Jan  6 12:00:32 monitor auth.err sshd[1887]: error: Could not get shadow information for NOUSER
Jan  6 12:00:32 monitor auth.info sshd[1887]: Failed password for invalid user c3os from 192.168.77.138 port 50054 ssh2
Jan  6 12:00:33 monitor auth.info sshd[1887]: Connection closed by invalid user c3os 192.168.77.138 port 50054 [preauth]
```

Is there some configuration to add to cloud-init to enable ntp?

**Comment by matclab** [comment]
This is using c3os-alpine-arm-rpi-v1.23.5-41.img.tar.xz image.

**Comment by mudler** [comment]
Hi! thanks for the feedback!

The alpine ARM image ain't yet completely functional, but ntpd should be functional. To configure the arm image see the docs here: https://docs.c3os.io/installation/raspberry/ 
TL;DR;

You need to copy the cloud-init file inside the `cloud-config` folder ( `cloud-config/<cloud-init>`) in the `COS_PERSISTENT` partition.

**Comment by matclab** [comment]
Thank you for the quick response.

I indeed followed the install instructions and the cloud-config is present (and taken into account):
```yaml
name: "Default deployment"
stages:     
   network:
     - name: "Configure NTP"
       ntp:
         pool:
           - 0.pool.ntp.org
           - 1.pool.ntp.org

     - if: '[ ! -f "/run/cos/recovery_mode" ]'
       name: "Setup k3s"
       environment_file: "/etc/systemd/system/k3s.service.env"
       environment:
         FOO: "bar"
       systemctl:
         start: 
         - k3s
     - commands:
       - |
            chmod 600 /etc/systemd/system/k3s.service.env
   initramfs:
   # Disable password, enable only ssh access
     - name: "Setup hostname"
       hostname: "monitor.clabaut.net"
     - if: '[ ! -d "/home/c3os" ]'
       name: "Ensure home directory is present"
       commands:
       - mkdir /home/c3os
       - chmod 755 /home/c3os
       - chown -R c3os /home/c3os
     - name: "Setup users"
       users:
        root:
          password: "c3os"
       authorized_keys:
        c3os: 
        - ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAICwfZvK1WCxwIlNvVSerMeN52w6DpYMdnTxMUf4LXs6K clabaut@loriot
       commands:
       - passwd -l c3os
     - name: "Ensure PVC dir exists"
       commands:
         -  mkdir -p /usr/local/prometheus-pvc /usr/local/grafana-pvc /usr/local/freebox-pvc
         -  chgrp 65534 /usr/local/prometheus-pvc /usr/local/grafana-pvc /usr/local/freebox-pvc /usr/local
         - chmod g+rwX /usr/local/prometheus-pvc /usr/local/grafana-pvc /usr/local/freebox-pvc /usr/local
```

I tried with and without the ntp section.

Do you know how  I can enable back the console login (scree+keyboard) for c3os user to better diagnose the problem?

**Comment by mudler** [comment]
here is the catch: the alpine image doesn't have systemd/systemctl so the `ntp` block won't work - and at the moment only the `initramfs` stage is functional (the boot and network openrc service are currently missing), which will be addressed soon.

You should be able to switch tty, otherwise there must be something wrong with the image - do you get the installation prompt with the QR code? that shouldn't happen, but should be enough to press enter in such case

**Comment by matclab** [comment]
I installed without VPN (no `c3os` config block) and with k3s only, and I indeed have a login prompt on the console, but according to the log file, the `c3os` account is locked (presumably because of the date not being set, but I'm not sure). `root` login doesn't work too.

I will try to inspect the image in order to understand if and how I can enable back `root` login on the console.

If you have some other ideas in order to connect to the live system I'll take it!

In the meanwhile, I'll probably try with the older opensuze image (I may be wrong, but I didn't find one in the last release).

**Comment by mudler** [comment]
The `c3os` account is locked because there is no default password, note that in the cloud-init config you provided: 

```yaml
   initramfs:
   # Disable password, enable only ssh access
     - name: "Setup hostname"
       hostname: "monitor.clabaut.net"
     - if: '[ ! -d "/home/c3os" ]'
       name: "Ensure home directory is present"
       commands:
       - mkdir /home/c3os
       - chmod 755 /home/c3os
       - chown -R c3os /home/c3os
     - name: "Setup users"
       users:
        root:
          password: "c3os"
       authorized_keys:
        c3os: 
        - ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAICwfZvK1WCxwIlNvVSerMeN52w6DpYMdnTxMUf4LXs6K clabaut@loriot
       commands:
       - passwd -l c3os
 ```

The `passwd -l c3os` command here will nuke any password and lock the account, so logging in from terminal won't work either. You should clean up that part and replace it with something like:

```yaml
stages:
   initramfs:
     - name: "Set user and password"
       users:
        c3os:
          passwd: "c3os"
```

which sets the user/pass to `c3os/c3os`.

**Comment by matclab** [comment]
Ho, you're right, I tried without the `passwd -l c3os`, but without creating the c3os password (I thought it was a default).
I still have:
```
Jan  6 12:03:43 monitor auth.warn login[2059]: invalid password for 'c3os' on 'tty1'
```
I will investigate and keep you informed.

**Comment by mudler** [comment]
This should be fixed in the next release :slightly_smiling_face:

**Comment by mudler** [comment]
Looks good now here, please try with the latest release again and reopen if necessary!
